### PR TITLE
1360335 - register host after network is up

### PR DESCRIPTION
**Pull request by jmrodri**
Move registration to after proceed with installation and only
after network is up.  If registration fails, exit. Fix prompt
so it doesn't show "Do not" for register_host.

### CODE REVIEW

**Comment by jmrodri** [comment]
## TEST
- Run installer for the first time before we configure the network.
- Choose to register host
- enter username and password
- verify `Register For Updates` displays `true`
- Then press 1 to proceed with installation
- Network should get configured, then we should successfully register.

**Comment by jmrodri** [comment]
If you choose to register for updates and it fails, we exit the installer.

**Comment by jmrodri** [comment]
https://bugzilla.redhat.com/show_bug.cgi?id=1360335
https://bugzilla.redhat.com/show_bug.cgi?id=1360855

File: hooks/lib/provisioning_wizard.rb [hunk 1]
```diff
@@ -254,20 +254,24 @@ def timezone
...
-      cmd = "subscription-manager register --username #{username} --password #{password} --auto-attach"
-      ret = system(cmd)
-      if ret.eql?(false)
-        say "<%= color('There was an error in registering this host!', :bad) %>"
-      else
-        say "<%= color('This host was successfully registered!', :good) %>"
-      end
-      @register_host = false
+      @portal_password = password
+      #cmd = "subscription-manager register --username #{username} --password #{password} --auto-attach"
```
Comment (jwmatthews) [inline hooks/lib/provisioning_wizard.rb]: Should we remove this commented out code, or do you want to leave it in?

Comment (jmrodri) [inline hooks/lib/provisioning_wizard.rb]: Nope, I'll remove it.

Comment (jmrodri) [inline hooks/lib/provisioning_wizard.rb]: Fixed, squashed, pushed.

**Comment by jwmatthews** [comment]
ACK

**Pull request merged**